<?xml version="1.0" encoding="utf-8"?>

<launch>
  <!-- Set the topics -->
  <arg name="input_map_topic"              default="/elevation_mapping_lidar/elevation_map_filtered"/>
  <arg name="input_fixed_frame"            default="odom"/>
  <arg name="input_base_frame"             default="base_link"/>
  <arg name="output_graph_topic"           default="/exploration_graph"/>
  <arg name="output_traversability_topic"  default="/traversability"/>
  <arg name="output_frontiers_topic"       default="/frontiers"/>
  
  <arg name="node_creation_distance"       default="1.0"/>
  <arg name="neighbor_distance"            default="2.0"/>
  <arg name="neighbour_path_distance"      default="5"/>
  
  <!-- Launch object detector -->
  <node pkg="world_modelling_cdt" type="world_modelling_cdt_node" name="world_modelling" output="screen">
    <param name="input_map_topic"              value="$(arg input_map_topic)"/>
    <param name="input_fixed_frame"            value="$(arg input_fixed_frame)"/>
    <param name="input_base_frame"             value="$(arg input_base_frame)"/>
    <param name="output_graph_topic"           value="$(arg output_graph_topic)"/>
    <param name="output_traversability_topic"  value="$(arg output_traversability_topic)"/>
    <param name="output_frontiers_topic"       value="$(arg output_frontiers_topic)"/>
    
    <param name="node_creation_distance"       value="$(arg node_creation_distance)"/>
    <param name="neighbor_distance"            value="$(arg neighbor_distance)"/>
    
    <param name="explored_space_length"       value="50.0"/>
    <param name="explored_space_resolution"   value="0.5"/>
    
    <param name="max_distance_to_search_frontiers"    value="5.0"/>
    <param name="distance_to_delete_frontier"         value="3.5"/>
    <param name="frontier_min_separation"             value="1.2"/>
    <param name="frontier_search_angle_resolution"    value="3"/>  <!-- Degrees -->
    <param name="frontier_search_angle"               value="200"/> <!-- Degrees -->

    <param name="trav_gradient_limit"                 value="0.3"/>
  
  </node>

</launch>
